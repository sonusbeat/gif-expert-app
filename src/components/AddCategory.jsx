import React from 'react'
import { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (event) => {
    setInputValue(event.target.value);
  };


  const handleSubmit = (event) => {
    event.preventDefault();

    if(inputValue.trim().length > 2) {
      setCategories(categories => [ inputValue, ...categories ]);
      setInputValue('');
    }

  };

  return (
    <form onSubmit={ handleSubmit }>
      <p style={{ display: "none" }}>{ inputValue }</p>
      <input
        className="form-control"
        type="text"
        value={ inputValue }
        onChange={ handleInputChange }
      />
    </form>
  )
}

export default AddCategory;

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}