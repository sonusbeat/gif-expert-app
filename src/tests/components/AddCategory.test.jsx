import AddCategory from '../../components/AddCategory';
import { shallow } from 'enzyme';

describe("Pruebas en el componente <AddCategory />", () => {
  const setCategories = () => {};

  test("Debe mostrar el componente correctamente", () => {
    const wrapper = shallow(<AddCategory setCategories={setCategories} />);

    expect(wrapper).toMatchSnapshot();
  });
});
