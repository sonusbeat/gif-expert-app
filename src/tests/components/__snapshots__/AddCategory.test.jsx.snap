// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AddCategory /> Debe mostrar el componente correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="mb-2"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
